solarextrapolation/utilities.py

```diff
@@ -0,0 +1,155 @@
+    
+    # If we have 2 bases, populate from the second base
+    if len(qua_input.unit.bases) > 1:
+        if qua_input.unit.bases[1].physical_type is u.m.physical_type:
+            length_unit = 1.0 * qua_input.unit.bases[1]
+            length_exponent = qua_input.unit.powers[1]
+            
+            # Convert to SI (meter here)
+            length_unit = length_unit.to(u.m)
+        elif qua_input.unit.bases[1].physical_type is u.radian.physical_type:
+            angle_unit = 1.0 * qua_input.unit.bases[1]
+            angle_exponent = qua_input.unit.powers[1]
+            
+            # Convert to SI (radian here)
+            angle_unit = angle_unit.to(u.radian)
+        
+    # Convert the incompatible base to the working units using the equivilence
+    if working_units.physical_type is u.m.physical_type:
+        angle_unit = angle_unit.to(working_units, equivalencies=equivalence)
+        
+        # Strip out the units, so the output doesn't have squared lenth units
+        #angle_unit = angle_unit.value # Kept in-case it causes bugs
+    elif working_units.physical_type is u.radian.physical_type:
+        length_unit = length_unit.to(working_units, equivalencies=equivalence)
+
+        # Strip out the units, so the output doesn't have squared length units
+        #length_unit = length_unit.value # Kept in-case it causes bugs
+    # The quantity to return
+    quantity =  value * length_unit ** length_exponent * angle_unit ** angle_exponent
+    # Change to the working unit if not dimensionless
+    if quantity.unit.physical_type is not (u.m / u.m).decompose().physical_type:
+        quantity.to(working_units)
+    return quantity.decompose()
+    
+
+def si_this_map_OLD(map):
+    """
+    Basic function to create a deep copy of a map but with all units in SI.
+    """
+    # Find out the value units and convert this and data to SI
+    units = 1.0 * u.Unit(map.meta['bunit']).to(u.Tesla) * u.Tesla
+    print units
+    data = deepcopy(map.data) * units.value
+    
+    # ATM I don't convert the x-axis and y-axis to SI
+    
+    # Modify the map header to reflect all these changes
+    meta = deepcopy(map.meta)
+    meta['bunit']   = units.unit
+    meta['datamax'] = data.max()
+    meta['datamin'] = data.min()
+    #meta['cdelt1'] = 0.504295 # Following modified if we convert x/y-axes
+    #meta['cdelt2'] = 0.504295
+    #meta['cunit1'] = 'arcsec'
+    #meta['cunit2'] = 'arcsec'
+    #meta['crpix1'] = data.shape[1] / 2.0 + 0.5, # central x-pixel
+    #meta['crpix2'] = data.shape[0] / 2.0 + 0.5, # cnetral y-pixel
+    #meta['CRVAL1'] = 0.000000
+    #meta['CRVAL2'] = 0.000000
+    
+    # Return the modified map
+    return mp.Map((data, meta))    
+
+
+def si_this_map(map):
+    """
+    Basic function to create a deep copy of a map but with all units in SI.
+    """
+    # Find out the value units and convert this and data to SI
+    units = 1.0 * u.Unit(map.meta['bunit']).to(u.Tesla) * u.Tesla
+    print units
```

Comment:
OLD function, to be removed once I have fully tested the second implementation, that SI's the length units too. I think that second part will be necessary in order to create a suitable dataset for all cuboid grid vector fields.